### ISSUE TITLE
Broken link in the fittedtemplate example

### ISSUE THREAD

**Issue by contsili** [fieldtrip/website#820]
[The fittedtemplate example documentation](fittedtemplate example) has a broken link. I believe [this is the correct link](https://github.com/markeroon/matlab-computer-vision-routines/blob/master/third_party/CPD2/core/cpd_register.m) is for `prepare_mesh_fittemplate` to work is

**Comment by contsili** [comment]
Why don't we include this function in FT by default?
EEGLAB does that with a lot of plugins.

**Comment by schoffelen** [comment]
hmm, I don't see where the link should be inserted in the website. Do you perhaps mean that it should be changed in ft_hastoolbox @contsili ?

**Comment by contsili** [comment]
This is the line with the broken link: ''This requires an external toolbox, which can be downloaded [here](https://sites.google.com/site/myronenko/research/cpd)''

**Comment by contsili** [comment]
> Do you perhaps mean that it should be changed in ft_hastoolbox @contsili ?

Yes we should also change `ft_hastoolbox`.

But my suggestion is more like: why don't we have a webpage in FT where we put all the plugins/external toolboxes. Then we avoid external links that we cannot control and might break in the future. See EEGLAB for inspiration:  https://sccn.ucsd.edu/eeglab/plugin_uploader/plugin_list_all.php

**Comment by schoffelen** [comment]
> This is the line in the website with the broken link: ''This requires an external toolbox, which can be downloaded {here}(https://sites.google.com/site/myronenko/research/cpd)''. This is a broken link

Yes I understand, but I could not find the link in the example page that you referenced in the issue.

**Comment by robertoostenveld** [comment]
we have this https://www.fieldtriptoolbox.org/faq/external/, but it is not automatically updated and quite out of date.

**Comment by contsili** [comment]
I tag @simonhomoelle who seems to be the one that first introduced CPD in FT.

I cannot find the CPD toolbox by Myrenko anywhere in the web. I only found it in C++ code: https://github.com/gadomski/cpd/tree/main

I see three options here: 

1. Use a built-in Matlab function that performs CPD: [pcregistercpd](https://nl.mathworks.com/help/vision/ref/pcregistercpd.html). We could use this instead.

2. Write our own function that does CPD. I think most of the recipe is here: https://github.com/markeroon/matlab-computer-vision-routines/blob/master/third_party/CPD2/core/cpd_register.m
 
3. Forget totally about CPD. It seems to not be important for MEG data analysis. I just stepped into it in the [fittedtemplate example documentation](https://www.fieldtriptoolbox.org/example/fittemplate/#method-2-on-the-basis-of-the-full-head-surface) which is just used as a second method.

To make all these changes we need to update ft_prepar_mesh, prepare_mesh_fittemplate, ft_hastoolbox, the [fittedtemplate example doc](https://www.fieldtriptoolbox.org/example/fittemplate/#method-2-on-the-basis-of-the-full-head-surface), and  https://www.fieldtriptoolbox.org/faq/external/. Quite some work I would say.

**Comment by schoffelen** [comment]
I would vote for 2), and put a copy of this file in fieldtrip/external/cpd, it's shared under the GNU2 licence so that would be fine. It would be nice to add a README in  external/cpd describing where the included file was obtained (i.e. linking to the github repo). You could perhaps sample some other directories in fieldtrip/external to get some inspiration.
